duckduckgo/chrome-zeroclickinfo
# fix domain matching

**Reviewer:**



## Description:

We're incorrectly sending the full url in the `domain` parameter to `abp.matches`. 
Instead we need to be sending the actual site domain. 
https://github.com/duckduckgo/chrome-zeroclickinfo/blob/c5214250dc9229d4964bd160819caf96929fa007/js/trackers.js#L224

I changed the following:
- passing the whole tab object into trackers.isTracker 
- updated trackers.js to use the siteDomain for our whitelist and easylist matching
- added some more isTracker tests to check some of the more complicated easylist filters
- started a test for the abp parser module. This needs to be expanded a lot which I plan to do in a later PR

## Steps to test this PR:

- Run unit tests. 
- The only test case I've found in the wild for this is on yahoo.com. 
-- without this fix you shouldn't see a `s.yimg.com` request being blocked
-- with this fix you should see one. It's correctly matching on this easylist entry `||yimg.com/rq/darla/$domain=yahoo.com`
- Test other popular sites

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
